tests/test_authority.py

```diff
@@ -123,3 +128,81 @@ class MockResponse(object):
         finally:  # MUST NOT let the previous test changes affect other test cases
             Authority._domains_without_user_realm_discovery = set([])
 
+
+_mocked_tenant_discovery_response = {
+    "authorization_endpoint": "https://contoso.com/placeholder",
+    "token_endpoint": "https://contoso.com/placeholder",
+    }
+
+@patch.multiple(
+    "msal.authority",
+    instance_discovery=DEFAULT,
+    tenant_discovery=Mock(return_value=_mocked_tenant_discovery_response),
+    )
+class TestMsalAuthorityBehaviorsWithoutKnownToDeveloperAuthorities(unittest.TestCase):
+
+    def test_a_known_to_microsoft_authority_should_skip_instance_discovery(self, **mocks):
+        msal.PublicClientApplication("id", authority="https://login.microsoftonline.com/common")
+        mocks["instance_discovery"].assert_not_called()
```

Comment:
Is it "known to microsoft" because you hardcode it somewhere? 

Not sure I agree with this test, because if a new alias is added, MSAL will get into SSO islands problem.

Reply:
> Is it "known to microsoft" because you hardcode it somewhere?

Yes. Historically, many ADAL/MSAL hardcodes a small set of hosts that are known to Microsoft. It was also known as "well known hosts" back then.

> Not sure I agree with this test, because if a new alias is added, MSAL will get into SSO islands problem.

Good point. In this test case, only the known-to-Microsoft check would be skipped, but the authority aliases i.e. instance metadata is still performed. Let me figure out how to rewrite this test case to show that. Stay tuned.

Reply:
OK, the test cases are updated to test two behaviors:
* "validation", also written as "authority_validation", but they really just mean "known-to-Microsoft validation" i.e. the Instance Discovery endpoint v1.0 behavior.
* The instance metadata acquisition